include/crm/common/results.h

```diff
@@ -326,6 +326,29 @@ enum pcmk_exec_status {
     PCMK_EXEC_MAX = PCMK_EXEC_NO_SECRETS, //!< Maximum value for this enum
 };
 
+/*!
+ * \enum pcmk_result_type_e
+ * \brief Types of Pacemaker result codes
+ *
+ * A particular integer can have different meanings within different Pacemaker
+ * result code families. It may be interpretable within zero, one, or multiple
+ * families.
+ *
+ * These values are useful for specifying how an integer result code should be
+ * interpreted in situations involving a generic integer value. For example, a
+ * function that can process multiple types of result codes might accept an
+ * arbitrary integer argument along with a \c pcmk_result_type_e argument that
+ * specifies how to interpret the integer.
+ */
+typedef enum pcmk_result_type_e {
+    pcmk_result_legacy        = 0,  //!< Legacy API function return code
+    pcmk_result_pcmk_rc       = 1,  //!< Standard Pacemaker return code
+    pcmk_result_crm_exitcode  = 2,  //!< Exit status code for tools and daemons
+    pcmk_result_ocf_exitcode  = 3,  //!< Exit status code for resource agents
+    pcmk_result_exec_status   = 4,  //!< Execution status
+    pcmk_result_bz2           = 5,  //!< Return code from bzlib
+} pcmk_result_type_t;
```

Comment:
@clumens , do you have any thoughts on a new enum vs reusing GQuark for this?

Reply:
That's an interesting idea and I think I like it.  GQuarks are made for identifying error domains, which is essentially what we are doing here.  Off the top of my head, I can't think of any reasons not to.

Reply:
I think the biggest downside is how to compare the value. I don't think we can use switch or `==`; we'd have to use `g_intern_string()` or something like that.

It also means the relevant public APIs will require glib. We already use GList and GHashTable elsewhere, so it's not that big of a deal, but I do like to minimize how much glib we expose.

Reply:
You should just be able to compare GQuarks.  They're just typedef'd to a guint32 (at least in this one version of the documentation I'm seeing - can't imagine that's changed).

Reply:
The doc says GQuarks are meant to be unique per string, so I think that would work. Let me know if we want to go that route. I guess it would require making the two existing GQuarks public (in name and in header) and then adding a couple of new ones.

The only thing I have in favor of an enum is that the meaning is especially clear (doxygen comment would solve that) and the values are constrained to what's defined.

Reply:
Another consideration is that the quark constants are actually function calls, so that's a good bit of overhead vs enums. Since we're only planning on using this in crm_error, I think that's OK. Hopefully it won't bite us later.

Let's just add one new one for legacy codes since that's all crm_error will need right now. It would be easy enough to add others later.

Reply:
We can get around this with `if` statements, but this is annoying.
```
results.c:51:9: error: case label does not reduce to an integer constant
   51 |         case PCMK_LEGACY_ERROR:
      |         ^~~~
results.c:55:9: error: case label does not reduce to an integer constant
   55 |         case PCMK_RC_ERROR:
      |         ^~~~
results.c:59:9: error: case label does not reduce to an integer constant
   59 |         case PCMK_EXITC_ERROR:
```

Reply:
It's also impossible to use these as initializers.
```
struct {    
    gboolean with_name;
    gboolean do_list;
    GQuark result_type; // How to interpret result codes
} options = {   
    .result_type = PCMK_LEGACY_ERROR,
}; 

In file included from ../include/crm/services.h:23,
                 from ../include/crm/lrmd.h:16,
                 from ../include/crm_internal.h:37,
                 from crm_error.c:10:
../include/crm/common/results.h:81:29: error: initializer element is not constant
   81 | #define PCMK_LEGACY_ERROR   pcmk_legacy_error_quark()
      |                             ^~~~~~~~~~~~~~~~~~~~~~~
crm_error.c:26:20: note: in expansion of macro 'PCMK_LEGACY_ERROR'
   26 |     .result_type = PCMK_LEGACY_ERROR,
      |                    ^~~~~~~~~~~~~~~~~
../include/crm/common/results.h:81:29: note: (near initialization for 'options.result_type')
   81 | #define PCMK_LEGACY_ERROR   pcmk_legacy_error_quark()
      |                             ^~~~~~~~~~~~~~~~~~~~~~~
crm_error.c:26:20: note: in expansion of macro 'PCMK_LEGACY_ERROR'
   26 |     .result_type = PCMK_LEGACY_ERROR,
      |                    ^~~~~~~~~~~~~~~~~
```

I'm almost finished (I think), but I'm going to hold off in case we want to rethink using the quarks for this. So far, it seems like it's 100% possible to use the quarks to identify the result code families, but:
1. It will make the code clunkier (can't use it in `switch` or static initializers), and
2. It requires function call overhead (probably not a big deal but it was noted above).

Reply:
Yeah, let's stick with enum

Reply:
I forgot to mention, enums don't need _e at the end or typedef, though you'll see both in legacy code. "enum X" is sufficient and ideal to indicate X is an enum.